CHANGES_NEXT_RELEASE

```diff
@@ -22,6 +22,8 @@
 - Fix: A bug fixed in URI param 'q', avoiding false matches when using string match operator (~=) that matches AND some operator that does not match.
```

Comment:
I have been having a closer look to this. The fixme is located in `equalMetadataValue()` function, which in secuence is called from `equalMetadata()` which is secuence is used in the logic to decide if an attribute values has changed and a notification has be triggered or not:

```
    /* In the case of simple value, we consider there is an actual change if one or more of the following are true:
     *
     * 1) the value of the attribute changed (see attrValueChanges for details)
     * 2) the type of the attribute changed (in this case, !attr.hasField(ENT_ATTRS_TYPE) is needed, as attribute
     *    type is optional according to NGSI and the attribute may not have that field in the BSON)
     * 3) the metadata changed (this is done checking if the size of the original and final metadata vectors is
     *    different and, if they are of the same size, checking if the vectors are not equal)
     */
      actualUpdate = (attrValueChanges(attr, caP, apiVersion) ||
                      ((caP->type != "") && (!attr.hasField(ENT_ATTRS_TYPE) ||
                                             getStringFieldF(attr, ENT_ATTRS_TYPE) != caP->type) ) ||
                      mdNew.nFields() != mdSize || !equalMetadata(md, mdNew));
```

So, not implementing the array/object cases in `equalMetadataValue()` as described in the FIXME will result in that part of the || be always true (note the `!`). In other words, any change in metadata compound is considered a change, although the new JSON could be structurally equal to the old one.

What we are doing in the same case for attributes? The same approach:

```
  if (caP->compoundValueP == NULL)
  {
    <the above code, which is for simple attribute values>
  }
  else
  {
      // FIXME P6: in the case of compound value, it's more difficult to know if an attribute
      // has really changed its value (many levels have to be traversed). Until we can develop the
      // matching logic, we consider actualUpdate always true.
      actualUpdate = true;
  }
```

Actually, we have an issue for that: https://github.com/telefonicaid/fiware-orion/issues/643

Thus, what we should do? I don't think we should solve this in this PR, but we can use this PR to "mark" the code in te appropiate places. In particular:

*  "// FIXME P6" above to "// FIXME #643 P6"
* "FIXME not yet, issue #1068 Support array and object in metadata value" -> "FIXME #643 P6: metadata array/object are now supported, but we haven't implemented yet the logic to compare compounds between them".

In addition, #643 will be expanded, to include a comment about this (I'll take care of that).